### ISSUE TITLE
Forem editor modals are not accessible

### ISSUE THREAD

**Issue by vaidehijoshi** [forem/forem#12427]
**Describe the bug**

The `/new` route on any Forem instance renders that Forem's editor. On the editor's sidebar, there are some basic tips that are meant to help the user, and those tips render in a modal (which opens on the editor page).

<img width="1295" alt="Screen Shot 2021-01-25 at 11 36 36 AM" src="https://user-images.githubusercontent.com/6921610/105756615-ad2dd480-5f01-11eb-81a3-214227eac164.png">

Those modals do not seem to be fully accessible -- once the modal opens, it appears that the focus does _not shift_ to the modal open on the page. For example, when the modal is opened, hitting the `escape` key does not close it...you need to tab a few times in order for the tab index to be "focused" on an element in the modal view. Only _then_ can you actually close the modal using the escape key.

There may be other a11y issues in these editor modals as well, but this is the most glaring issue.

**To Reproduce**

Click on the below gif to watch a loom recording of this bug in action:

[![gif of editor bug](https://cdn.loom.com/sessions/thumbnails/eb891189427a4072999e312d0e4003bb-with-play.gif)](https://www.loom.com/share/eb891189427a4072999e312d0e4003bb)

(Or just [click here](https://www.loom.com/share/eb891189427a4072999e312d0e4003bb) to watch the video).

**Expected behavior**

I would expect to be able to tab through the editor's modal and be able to close it once it is opened. A user should not know that they have to keep tabbing until they are in the right "context" to be able to dismiss a modal without using a mouse to do so :)

**Comment by joaoGabriel55** [comment]
Hello @vaidehijoshi, I would like to work on this.

**Comment by vaidehijoshi** [comment]
@joaoGabriel55 Hi there! Thanks for the offer :) This issue has the `Forem team` tag on it, which means that it is an internal task that will be completed by someone on the Forem core team. 

I'd recommend taking a look at our [Contributing Guide](https://docs.forem.com/contributing/forem/). All of our issues labeled `ready for dev` and `type: bug` are generally up for grabs, so anyone from the community can work on them 👍🏽

**Comment by ludwiczakpawel** [comment]
The `escape` event has been [added](https://github.com/forem/forem/pull/12428). Focus-trap - nope: there's a community [member working on that functionality elsewhere](https://github.com/forem/forem/pull/12366).

@vaidehijoshi you can make a call if `escape` event is enough to close this issue and open separate one for focus-trap or should we handle it separately (moved to _[Blocked](https://github.com/orgs/forem/projects/14#column-12643450)_?)

**Comment by vaidehijoshi** [comment]
@ludwiczakpawel Awesome! Since this issue is tied to both the escape event and the focus trap on the modal (in progress in https://github.com/forem/forem/pull/12366), let's add this to the blocked column temporarily.

Once https://github.com/forem/forem/pull/12366 is merged, let's confirm and check that the changes introduced by _that_ PR also fix the focus trap problem surfaced in this issue. If so, we can close this issue out at that point! 🎉

**Comment by aitchiss** [comment]
@vaidehijoshi @ludwiczakpawel #12366 has been successfully merged but it only addresses focus management in the onboarding modal. We can now unblock this issue and generalise the new code to make sure all of our modals are accessible 👍
